### DISCUSSION TITLE
Using SpaCy LLM to generate regular text

### DISCUSSION THREAD

**Discussion by harryjulian** [explosion/spacy-llm#314] [Q&A] [answered]
I understand it might not necessarily be the point of the repo, but I'm working on a project where I want to use a few `spacy-llm` defined tasks (NER and text classification) as well as getting the model to generate text. What's the best way to do this within the framework? I'm trying to avoid having multiple ways of loading/calling the same model. I'm aiming to use the LangChain CTransformers integration.

**Comment by rmitsch** [comment]
Hi @harryjulian, this is more of a discussion than an issue, let me convert this.

**Comment by rmitsch** [comment, marked as answer]
You could have a look at the implementation of `spacy.Summarization.v1`. In a nutshell you'd have to define your own task (you can copy the implementation of the summarization task for the most part) and your own prompt - instructing the LLM what text to generate. This text you can then store in e. g. your docs `_` [attribute](https://spacy.io/api/doc#set_extension).
